sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -772,12 +772,17 @@ object SQLConf {
         " produced by a builtin function such as to_char or CAST")
       .version("4.0.0")
       .stringConf
-      .checkValue(CollationFactory.isValidCollation,
+      .checkValue(
+        collationName => {
+          try {
+            CollationFactory.fetchCollation(collationName)
+            true
+          } catch {
+            case e: SparkException if e.getErrorClass == "COLLATION_INVALID_NAME" => false
+          }
+        },
```

Comment:
Why are we making this change as well? Why not keep the `isValidCollation` predicate?

Reply:
`isValidCollation` would contain exception catching as part of its control flow, just as this approach does. Our reasoning was that it's better to scope this approach only here and not leave it open for use in other places.
cc @dbatomic

Reply:
Resolving the discussion, we removed `isValidCollation` from `CollationFactory` and chose to scope it to only here as it's logic of catching exception as part of control flow is not a good approach for general-purpose utility function.